lib/pluggy_elixir/http_adapter.ex

```diff
@@ -0,0 +1,21 @@
+defmodule PluggyElixir.HttpAdapter do
+  @moduledoc """
+  The HTTP Adapter specification.
+
+  An HttpAdapter is a module that perform HTTP requests and put response into
+  `PluggyElixir.HttpAdapter.Response` struct.
+  """
+
+  alias PluggyElixir.HttpAdapter.Response
+
+  @type url :: binary()
+  @type param :: binary() | [{binary() | atom(), param()}]
+  @type query() :: [{binary() | atom(), param()}]
+  @type adapter_response :: {:ok, Response.t()} | {:error, binary()}
+
+  @doc "Perform a HTTP request with GET method"
```

Comment:
Should we add an example of `get` function call here in the doc?

Reply:
It would be nice, but we can do it in the future